python/tank/util/filesystem.py

```diff
@@ -451,3 +453,50 @@ def get_unused_path(base_path):
         if not os.path.exists(path):
             break
     return path
+
+
+@with_cleared_umask
+@contextmanager
+def auto_created_yml(path):
+    """
+    A context manager for opening/closing an auto-generated yaml file.
+
+    - clears umask
+    - any existing files will be removed
+    - the given path will be open for writing in text mode
+    - a standard header will be added
+
+    Usage example::
+
+        with filesystem.auto_created_yml(yaml_path) as fh:
+            fh.write("foobar: blah")
+
+        # fh is automatically closed upon exiting the context
+
+    :param path: path to yml file to open for writing
+    :return: file handle.
+    """
+    fh = None
+
+    try:
```

Comment:
Use the `with`, Luke!!

```
log.debug(...)
safe_delete_path(...)
with open(path, "w+"):
   fh.write(...)
   yield fh
   fh.write(...)
```

This code is functionaly the same, but much easier to understand. Also less error prone because no need to test for fh.

Reply:
In retrospec tough, wouldn't you want to delete the file if an exception was raised?

Reply:
Fair enough.